File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        product_total = 1
        zero_count = 0
        solution = []
        for num in nums:
            if num == 0:
                zero_count += 1
            else:
                product_total *= num
        if zero_count >= 2:
            for num in nums:
                solution.append(0)
        elif zero_count == 1:
            for num in nums:
                if num != 0:
                    solution.append(0)
                else:
                    solution.append(product_total)
        else:
            for num in nums:

                solution.append(product_total // num)

        return solution

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int> numberCount;
        vector<int> frequentList;

        for (int num: nums) {
            if (numberCount.find(num) == numberCount.end()) {
                numberCount[num] = 1;
            } else {
                auto it = numberCount.find(num);
                ++it->second;
            }
        }

        int highestFirst = numberCount.begin()->first;
        int highestSecond = numberCount.begin()->second;
        while (k > 0) {
            for (const auto &pair: numberCount) {
                bool alreadyInList = false;
                for (int i = 0; i < frequentList.size(); ++i) {
                    if (pair.first == frequentList[i]) {
                        alreadyInList = true;
                    }
                }

                if (!alreadyInList && (pair.second > highestSecond)) {
                    highestFirst = pair.first;
                    highestSecond = pair.second;
                }
            }

            frequentList.emplace_back(highestFirst);
            highestSecond = -1001;

            --k;
        }

        return frequentList;


    }

};
